pkg/cmd/provisioning/identity_provider.go

```diff
@@ -0,0 +1,335 @@
+					Value: awssdk.String(bucketTagValue),
+				},
+			},
+		},
+	})
+	if err != nil {
+		return errors.Wrapf(err, "Failed to tag the bucket %s", bucketName)
+	}
+
+	discoveryDocumentJSON := fmt.Sprintf(discoveryDocumentTemplate, issuerURL, issuerURL, keysURI)
+	_, err = client.PutObject(&s3.PutObjectInput{
+		ACL:     awssdk.String("public-read"),
+		Body:    awssdk.ReadSeekCloser(strings.NewReader(discoveryDocumentJSON)),
+		Bucket:  awssdk.String(bucketName),
+		Key:     awssdk.String(discoveryDocumentURI),
+		Tagging: awssdk.String(fmt.Sprintf("%s/%s=%s", ccoctlAWSResourceTagKeyPrefix, infraName, ownedCcoctlAWSResourceTagValue)),
+	})
+	if err != nil {
+		return errors.Wrapf(err, "Failed to upload discovery document in the S3 bucket %s", bucketName)
+	}
+	log.Printf("OpenID Connect discovery document in the S3 bucket %s at %s updated", bucketName, discoveryDocumentURI)
+
+	err = saveToFile("discovery document", filepath.Join(targetDir, "oidc-configuration"), []byte(discoveryDocumentJSON))
+	if err != nil {
+		return err
+	}
+
+	jwks, err := buildJsonWebKeySet(publicKeyPath)
+	if err != nil {
+		return errors.Wrap(err, "Failed to build JSON web key set from the public key")
+	}
+
+	_, err = client.PutObject(&s3.PutObjectInput{
+		ACL:     awssdk.String("public-read"),
+		Body:    awssdk.ReadSeekCloser(bytes.NewReader(jwks)),
+		Bucket:  awssdk.String(bucketName),
+		Key:     awssdk.String(keysURI),
+		Tagging: awssdk.String(fmt.Sprintf("%s/%s=%s", ccoctlAWSResourceTagKeyPrefix, infraName, ownedCcoctlAWSResourceTagValue)),
+	})
+
+	if err != nil {
+		return errors.Wrapf(err, "Failed to upload JSON web key set (JWKS) in the S3 bucket %s", bucketName)
+	}
+	log.Printf("JSON web key set (JWKS) in the S3 bucket %s at %s updated", bucketName, keysURI)
+
+	err = saveToFile("JSON web key set (JWKS)", filepath.Join(targetDir, "keys.json"), jwks)
+	if err != nil {
+		return err
+	}
+
+	oidcProviderList, err := client.ListOpenIDConnectProviders(&iam.ListOpenIDConnectProvidersInput{})
+	if err != nil {
+		return errors.Wrap(err, "Failed to fetch list of Identity Providers")
+	}
+
+	var providerARN string
+	for _, provider := range oidcProviderList.OpenIDConnectProviderList {
+		ok, err := isExistingIdentifyProvider(client, *provider.Arn, infraName)
+		if err != nil {
+			return errors.Wrapf(err, "Failed to check existing Identity Provider %s", *provider.Arn)
+		}
+
+		if ok {
+			providerARN = *provider.Arn
+			log.Printf("Existing Identity Provider found with ARN: %s", providerARN)
+			break
+		}
+	}
+
+	if len(providerARN) == 0 {
+		oidcOutput, err := client.CreateOpenIDConnectProvider(&iam.CreateOpenIDConnectProviderInput{
+			ClientIDList: []*string{
+				awssdk.String("openshift"),
+				awssdk.String("sts.amazonaws.com"),
+			},
+			ThumbprintList: []*string{
+				awssdk.String("A9D53002E97E00E043244F3D170D6F4C414104FD"), // root CA thumbprint for Amazon S3 (DigiCert)
```

Comment:
I'd rather fetch this dynamically from the bucket we just created (in the event that GovCloud or AWS China is using different CAs).